docs/dev_guide/frequently-asked-questions-faq.rst

```diff
@@ -162,6 +163,95 @@ expected results. Instead, exclude CUB library implementation source code from
 your migration by adding ``--in-root-exclude=<path to CUB library source code>``
 to your migration command.
 
+How do I fix the hang issue of sycl code caused by work group level synchronization like group barrier used in conditional statement?
+*************************************************************************************************************************************
+
+In SYCL code, if there is calling of synchronization API in control flow statements

```

Comment:
If synchronization API in control flow statements like a conditional statement and loop statement are called in SYCL code, you may encounter a runtime hang issue.
The basic idea to fix the hang issue is to ensure that each synchronization API is
either reached by all work items of a workgroup, or skipped by all the work items of
a workgroup.

Here are two examples of how to fix: 

In the first example, the synchronization API group barrier(nd_item.barrier()) is called
inside an if block. The evaluation results of the conditional statement differs in
each work item so not all work items can reach the synchronization API.

Reply:
has fixed

Reply:
I made a change
"The evaluation results of the conditional statement differs in" ==> "The evaluation results of the conditional statement differ in"
please double check. Thanks